packages/react-dom/src/events/__tests__/ChangeEventPlugin-test.internal.js

```diff
@@ -11,212 +11,219 @@
-    ReactTestUtils.SimulateNative.click(input);
-
+    setUntrackedChecked.call(node, false);
+    node.dispatchEvent(new Event('click', {bubbles: true, cancelable: true}));
     expect(called).toBe(1);
-  });
+    expect(node.checked).toBe(true);
 
-  it('should catch setting the value programmatically', () => {
-    var input = ReactTestUtils.renderIntoDocument(
-      <input type="text" defaultValue="foo" />,
-    );
-
-    input.value = 'bar';
-    expect(getTrackedValue(input)).toBe('bar');
+    setUntrackedChecked.call(node, true);
+    node.dispatchEvent(new Event('click', {bubbles: true, cancelable: true}));
+    expect(called).toBe(2);
+    expect(node.checked).toBe(false);
   });
 
-  it('should not fire change when setting the value programmatically', () => {
+  fit('should not fire change setting the value programmatically', () => {
     var called = 0;
 
     function cb(e) {
-      called += 1;
+      called++;
       expect(e.type).toBe('change');
     }
 
-    var input = ReactTestUtils.renderIntoDocument(
-      <input type="text" onChange={cb} defaultValue="foo" />,
+    var input = ReactDOM.render(
+      <input type="text" defaultValue="foo" onChange={cb} />,
+      container,
     );
 
     input.value = 'bar';
-    ReactTestUtils.SimulateNative.change(input);
+    input.dispatchEvent(new Event('input', {bubbles: true, cancelable: true}));
     expect(called).toBe(0);
 
-    setUntrackedValue(input, 'foo');
-    ReactTestUtils.SimulateNative.change(input);
+    setUntrackedValue.call(input, 'foo');
+    input.dispatchEvent(new Event('input', {bubbles: true, cancelable: true}));
 
     expect(called).toBe(1);
+    expect(input.value).toBe('foo');
+
+    input.dispatchEvent(new Event('input', {bubbles: true, cancelable: true}));
+    expect(called).toBe(1);
   });
 
-  it('should not fire change when setting checked programmatically', () => {
+  fit('should not fire change when setting checked programmatically', () => {
     var called = 0;
 
     function cb(e) {
-      called += 1;
+      called++;
       expect(e.type).toBe('change');
     }
 
-    var input = ReactTestUtils.renderIntoDocument(
+    var input = ReactDOM.render(
       <input type="checkbox" onChange={cb} defaultChecked={true} />,
+      container,
     );
 
+    //set programmatically
     input.checked = true;
-    ReactTestUtils.SimulateNative.click(input);
+
+    setUntrackedChecked.call(input, false);
+    input.dispatchEvent(new Event('click', {bubbles: true, cancelable: true}));
     expect(called).toBe(0);
 
-    input.checked = false;
-    setTrackedValue(input, undefined);
-    ReactTestUtils.SimulateNative.click(input);
+    setUntrackedChecked.call(input, true);
+    input.dispatchEvent(new Event('click', {bubbles: true, cancelable: true}));
 
     expect(called).toBe(1);
   });
 
-  it('should unmount', () => {
-    var container = document.createElement('div');
+  fit('should catch setting the value programmatically', () => {
```

Comment:
I don't understand what this test is trying to test. What is the thinking behind it?

Reply:
I think it should test that the tracked value updates when you set it from js. The test should normally set the value then trigger and event and assert that the event doesn't fire

Reply:
Sorry, I meant that I don't understand what the *new* test is doing (since it's not testing setting the field at all).